mmdet/apis/inference.py

```diff
@@ -54,31 +54,50 @@ def init_detector(
         config.model.backbone.init_cfg = None
     config.model.train_cfg = None
     model = build_detector(config.model)
+    model.dataset_meta = {}
     if checkpoint is not None:
         checkpoint = load_checkpoint(model, checkpoint, map_location='cpu')
         # Weights converted from elsewhere may not have meta fields.
         checkpoint_meta = checkpoint.get('meta', {})
+
         # save the dataset_meta in the model for convenience
         if 'dataset_meta' in checkpoint_meta:
             # mmdet 3.x
-            model.dataset_meta = checkpoint_meta['dataset_meta']
+            dataset_meta = checkpoint_meta['dataset_meta']
```

Comment:
shall we directly use `model.dataset_meta=copy.deepcopy(checkpoint_meta['dataset_meta'])`

Reply:
There should be no need for deepcopy.